gpflow/conditionals/uncertain_conditionals.py

```diff
@@ -48,7 +48,7 @@ def uncertain_conditional(Xnew_mu: tf.Tensor,
 
     pXnew = Gaussian(Xnew_mu, Xnew_var)
 
-    num_data = Xnew_mu.shape[0]  # number of new inputs (N)
+    num_data = tf.shape(Xnew_mu)[0]  # number of new inputs (N)
     num_ind = q_mu.shape[0]  # number of inducing points (M)
     num_func = q_mu.shape[1]  # output dimension (D)
```

Comment:
should these by dynamic shapes as well ?

Reply:
Within the context of a defined model no, I agree that M&D are known, but if you were interested in reusing the function with generic tensors then maybe yes. Happy to revert.

Reply:
let me rephrase: is there anything we would *loose* by having these be `num_ind, num_func = tf.unstack(tf.shape(q_mu), num=2, axis=0)` as well?